src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpIdentifiersShouldNotContainUnderscores.Fixer.cs

```diff
@@ -13,5 +14,18 @@ namespace Microsoft.CodeQuality.CSharp.Analyzers.ApiDesignGuidelines
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
     public sealed class CSharpIdentifiersShouldNotContainUnderscoresFixer : IdentifiersShouldNotContainUnderscoresFixer
     {
+        protected override string GetNewName(string name)
+        {
+            string result = RemoveUnderscores(name);
+            if (result.Length == 0)
+                return string.Empty;
+
+            if (!SyntaxFacts.IsValidIdentifier(result))
```

Comment:
From the tests you have added it seems that this isn't checking that the identifier is not matching a reserved keyword so you will need to tweak it. That's the reason some tests (e.g. CA1707_ForMemberparameters_CSharp) are failing.

Reply:
@Evangelink Shouldn't the identifier after removing the underscores become invalid if and only if it's a reserved word?

Reply:
I'm unsure, but I *think* the "ForMemberparameters" test is failing because the call to `RenameSymbolAsync` on an interface parameter doesn't rename the parameter in an implementing class. Is that the expected behavior from `RenameSymbolAsync`?

Reply:
Not sure to understand your comment. In one of your test, you have a parameter `int_`, when reaching this line it is now `int` so I was expecting this method to return false resulting in `@int` which isn't the case.

Reply:
> 
> 
> I'm unsure, but I _think_ the "ForMemberparameters" test is failing because the call to `RenameSymbolAsync` on an interface parameter doesn't rename the parameter in an implementing class. Is that the expected behavior from `RenameSymbolAsync`?

I have seen this behavior in some other PR, that's not what I was expecting but yes it is the case.

Reply:
> you have a parameter int_, when reaching this line it is now int so I was expecting this method to return false resulting in @int which isn't the case.

```csharp
public class implementI : I
{
    public void M(int int_)
```

@Evangelink You mean the above parameter right?

The problem with it not being changed to `@int` is because the analyzer don't report for it.
The analyzer reported for the parameter in the interface `I`, which was indeed renamed to @int

Reply:
My bad I thought I was on a different context. You are right the only "problem" is linked to the missing rename for implementations.